src/libs/ck-libs/NDMeshStreamer/NDMeshStreamer.h

```diff
@@ -144,21 +175,25 @@ class MeshStreamer : public CBase_MeshStreamer<dtype, RouterType> {
   bool useCompletionDetection_;
   CompletionDetector *detectorLocalObj_;
   virtual int copyDataItemIntoMessage(
-              MeshStreamerMessage<dtype> *destinationBuffer,
-              const void *dataItemHandle, bool copyIndirectly = false);
-  void insertData(const void *dataItemHandle, int destinationPe);
-  void broadcast(const void *dataItemHandle, int dimension,
-                 bool copyIndirectly);
+              MeshStreamerMessageV *destinationBuffer,
+              const DataItemHandle<dtype> *dataItemHandle, bool copyIndirectly = false);
+  virtual int copyDataIntoMessage(
+              MeshStreamerMessageV *destinationBuffer,
+              char *dataHandle, size_t size, CkArrayIndex index);
+  void insertData(const DataItemHandle<dtype> *dataItemHandle, int destinationPe);
+  void storeMessageData(int destinationPe,
+                    const Route& destinationCoordinates,
+                    char *data, size_t size,CkArrayIndex);
   void storeMessage(int destinationPe,
                     const Route& destinationCoordinates,
-                    const void *dataItem, bool copyIndirectly = false);
+                    const DataItemHandle<dtype> *dataItem, bool copyIndirectly = false);
 
   void ctorHelper(int maxNumDataItemsBuffered, int numDimensions,
                   int *dimensionSizes, int bufferSize,
                   bool yieldFlag, double progressPeriodInMs);
 
 public:
-
+std::map<int,int> histogram;
```

Comment:
Add a comment describing what the key/val pairs are

Reply:
That line is actually something I forgot about when cleaning it up, I'll get rid of it.